aioircd.py

```diff
@@ -285,7 +284,23 @@ async def JOIN(self, args):
 
     @command
     async def PRIVMSG(self, args):
-        pass
+        if not args or args[0] == "":
+            raise ErrNoRecipient("PRIVMSG")
+        user,*mess = args
+        message = " ".join(mess)
+        if not message or not message.startswith(":") or len(message)<2:
+            raise ErrNoTextToSend()
+        if user.startswith("#"):
+            chann = self.local.channels.get(user)
+            if chann == None:
+                raise ErrNoSuchChannel(user)
+            await chann.send_all(f":{self.nick} PRIVMSG {user} :{message[1:]}")
+        else:
+            receiver = self.local.users.get(user)
+            if receiver == None:
```

Comment:
Do not use `==` when comparing to None, use `is`. By the way, as self.local.users values are User object, truthy value, we can simply `if not receiver`. Or taking advantage of walrus `if not receiver := self.local.users.get(user): exception`